pkg/compactor/lifecycle.go

```diff
@@ -0,0 +1,11 @@
+package compactor
+
+import (
+	"context"
+
+	"github.com/cortexproject/cortex/pkg/ring"
+)
+
+func (c *Compactor) OnRingInstanceHeartbeat(_ *ring.Lifecycler, ctx context.Context) {
```

Comment:
I would do similar to the basicLifecycler.

Create the generic delegate for the ring and make compactor use it.